servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/HttpResponseFactoryStatusCodeTest.java

```diff
@@ -0,0 +1,120 @@
+import io.servicetalk.http.api.HttpResponseFactory;
+import io.servicetalk.http.api.HttpResponseMetaData;
+import io.servicetalk.http.api.HttpResponseStatus;
+import io.servicetalk.http.api.StreamingHttpResponse;
+import io.servicetalk.http.api.StreamingHttpResponseFactory;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static io.servicetalk.http.api.DefaultHttpHeadersFactory.INSTANCE;
+import static org.junit.Assert.assertEquals;
+
+public class HttpResponseFactoryStatusCodeTest {
+    @Rule
+    public final Timeout timeout = new ServiceTalkTestTimeout();
+    private static List<HttpResponseStatus> expectedStatuses;
+
+    @BeforeClass
+    public static void before() {
+        expectedStatuses = IntStream.range(100, 600)
+                .mapToObj(code -> HttpResponseStatus.of(Integer.toString(code)))
+                .filter(status -> !"unknown".equals(status.reasonPhrase()))
+                .collect(Collectors.toList());
+    }
+
+    @Test
+    public void testStatusCode() {
+        final HttpResponseFactory httpResponseFactory =
+                new DefaultHttpResponseFactory(INSTANCE, BufferAllocators.DEFAULT_ALLOCATOR);
+
+        List<HttpResponseStatus> responseStatuses =
+                buildHttpStatuses(httpResponseFactory, HttpResponseFactory.class, HttpResponse.class);
+
+        assertCodeAndReason(expectedStatuses, responseStatuses);
+    }
+
+    @Test
+    public void testBlockingStatusCode() {
+
+        final BlockingStreamingHttpResponseFactory blockingStreamingHttpResponseFactory =
+                new DefaultBlockingStreamingHttpResponseFactory(INSTANCE, BufferAllocators.DEFAULT_ALLOCATOR);
+
+        List<HttpResponseStatus> responseStatuses =
+                buildHttpStatuses(blockingStreamingHttpResponseFactory,
+                        BlockingStreamingHttpResponseFactory.class, BlockingStreamingHttpResponse.class);
+
+        assertCodeAndReason(expectedStatuses, responseStatuses);
+    }
+
+    @Test
+    public void testStreamingStatusCode() {
+
+        final StreamingHttpResponseFactory streamingHttpResponseFactory =
+                new DefaultStreamingHttpResponseFactory(INSTANCE, BufferAllocators.DEFAULT_ALLOCATOR);
+
+        List<HttpResponseStatus> responseStatuses =
+                buildHttpStatuses(streamingHttpResponseFactory,
+                        StreamingHttpResponseFactory.class, StreamingHttpResponse.class);
+
+        assertCodeAndReason(expectedStatuses, responseStatuses);
+    }
+
+    private static void assertCodeAndReason(final List<HttpResponseStatus> expectedStatuses,
+                                            final List<HttpResponseStatus> actualStatuses) {
+
+        for (HttpResponseStatus expectedStatus : expectedStatuses) {
```

Comment:
You can simplify this by using hamcrest matchers:

```java
MatcherAssert.assertThat("unexpected statuses", actualStatuses,
        Matchers.containsInAnyOrder(expectedStatuses.toArray()));
```

Reply:
Thanks! Done